client/src/modules/locations/country/country.js

```diff
@@ -2,91 +2,93 @@ angular.module('bhima.controllers')
   .controller('CountryController', CountryController);
 
 CountryController.$inject = [
-  'LocationService', 'util', 'NotifyService',
+  'LocationService', 'util', 'LocationService',
+  'NotifyService', '$uibModal', 'ModalService',
 ];
 
-function CountryController(locationService, util, Notify) {
+function CountryController(locationService, util, Location, Notify, $uibModal, Modal) {
   const vm = this;
   vm.session = {};
   vm.view = 'default';
 
   // bind methods
-  vm.create = create;
-  vm.submit = submit;
-  vm.update = update;
-  vm.cancel = cancel;
   vm.countryLength = util.length45;
 
   // fired on startup
   function startup() {
     // start up loading indicator
     vm.session.loading = true;
-
     // load Country
-    refreshCountrys();
-  }
-
-  function cancel() {
-    vm.view = 'default';
-  }
-
-  function create() {
-    vm.view = 'create';
-    vm.country = {};
+    refreshCountries();
   }
 
   vm.messages = {
     country : locationService.messages.country,
   };
 
-  /** load countries on startup */
-  locationService.countries()
-    .then((countries) => {
-
-      // bind the countries to the view for <select>ion
-      vm.countries = countries;
-
+  // refresh the displayed Countrys
+  function refreshCountries() {
+    return locationService.countries({ detailed : 1 }).then((countries) => {
+      vm.gridOptions.data = countries;
+      vm.session.loading = false;
       // make sure that we are showing the proper message to the client
       vm.messages.country = (countries.length > 0)
         ? locationService.messages.country
         : locationService.messages.empty;
     });
-
-
-  // switch to update mode
-  // data is an object that contains all the information of a country
-  function update(data) {
-    vm.view = 'update';
-    vm.country = data;
   }
 
-  // refresh the displayed Countrys
-  function refreshCountrys() {
-    return locationService.countries({ detailed : 1 }).then((data) => {
-      vm.countries = data;
-      vm.session.loading = false;
+  vm.createUpdateModal = (selectedCountry = {}) => {
+    return $uibModal.open({
+      templateUrl : 'modules/locations/country/modal/createUpdate.html',
+      controller : 'CreateUpdateCountryController as ModalCtrl',
+      resolve : { data : () => selectedCountry },
+    }).result.then(result => {
+      if (result) refreshCountries();
     });
```

Comment:
If you click "cancel", a warning will probably appear in the console since you don't have  "catch" statement here.  You can fix this by adding `.catch(angular.noop)`.